src/cobald/composite/weighted.py

```diff
@@ -41,12 +44,25 @@ def utilisation(self):
     def allocation(self):
         try:
             return (
-                sum(child.allocation * child.supply for child in self.children)
-                / self.supply
+                sum(
+                    child.allocation * getattr(child, self._weight)
+                    for child in self.children
+                )
+                / self._total_weight
             )
         except ZeroDivisionError:
             return 1.0
 
-    def __init__(self, *children: Pool):
+    @property
+    def _total_weight(self):
+        return sum(getattr(child, self._weight) for child in self.children)
+
+    def __init__(self, *children: Pool, weight="supply"):
+        assert weight in (
+            "supply",
+            "utilisation",
+            "allocation",
```

Comment:
I am a bit afraid of using strings here. But I don't see another option.

Reply:
We could you use an enum instead?

Reply:
For the time being, leave it as is :)